File: JH/DAY12/10799.cpp
#include <bits/stdc++.h>
using namespace std;

stack<char>st;

int main(){


    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    string s;
    cin >> s;

    int cnt = 0;

    for (int i = 0; i <s.length() ; ++i) {
        if(s[i] == '('){
            st.push(s[i]);
        }else{
            if(s[i-1]=='('){
                st.pop();
                cnt+=st.size();
            }else{
                if(st.size()!=0){
                    st.pop();
                    cnt++;
                }
            }
        }
    }
    cout << cnt << "\n";
}

File: SY/stack/10799.py
li=list(input())
cnt=0
stack=[]
for i in range(len(li)):
	if li[i]=='(':
		stack.append(li[i])
	if li[i]==')':
		if li[i-1]=='(':
			stack.pop()
			cnt+=len(stack)
		elif li[i-1]==')':
			stack.pop()
			cnt+=1
print(cnt)